Panic Integer Divide by Zero when dividePartitionsBetweenConsumers

I have 1 consumer and what might be a finicky zookeeper connection which can cause a panic with a divide by zero.

Log output (AFAIK, zookeeper is up, else I'd have bigger problems, however this client is behind a VPN):
```
2015/07/03 22:32:29 Failed to connect to zoo1:2181: dial tcp 10.129.196.49:2181: i/o timeout
2015/07/03 22:32:31 Failed to connect to zoo2:2181: dial tcp 10.129.196.11:2181: i/o timeout
2015/07/03 22:32:32 read tcp 10.129.196.49:2181: i/o timeout
2015/07/03 22:32:33 Failed to connect to zoo2:2181: dial tcp 10.129.196.11:2181: i/o timeout
panic: runtime error: integer divide by zero
[signal 0x8 code=0x7 addr=0x36bafd pc=0x36bafd]

goroutine 544 [running]:
github.com/wvanbergen/kafka/consumergroup.dividePartitionsBetweenConsumers(0x1005fc8, 0x0, 0x0, 0xc208dd2200, 0x20, 0x20, 0x0)
	/Users/nimi/go/src/github.com/wvanbergen/kafka/consumergroup/utils.go:39 +0x16d
github.com/wvanbergen/kafka/consumergroup.(*ConsumerGroup).topicConsumer(0xc2081a8000, 0xafae10, 0xa, 0xc208162f60, 0xc208163320, 0xc208dfdce0)
	/Users/nimi/go/src/github.com/wvanbergen/kafka/consumergroup/consumer_group.go:310 +0x7c3
created by github.com/wvanbergen/kafka/consumergroup.(*ConsumerGroup).topicListConsumer
	/Users/nimi/go/src/github.com/wvanbergen/kafka/consumergroup/consumer_group.go:260 +0x3f6
```

Which points to [here](https://github.com/wvanbergen/kafka/blob/f6744f8c92c4257e043ff72e9c80a77196fe427c/consumergroup/utils.go#L39)

```go
sort.Sort(partitions)
sort.Sort(consumers)

n := plen / clen
if plen%clen > 0 {
	n++
}
```

I'm assuming (haven't done any digging) that we see no consumers (because zookeeper madness?) and try to divide partitions among zero consumers. Given the ZooKeeper connection should eventually comeback, what should be the right course of action here? I think it would be ideal for the consumer to do no work until the connection with ZK stabilizes/comes back.

I ran into this overnight last night.  Mine was in a slightly different context, during a rebalance operation while system was idle:


<pre>[Sarama] 2015/07/13 02:02:13 client/metadata fetching metadata for all topics from broker 192.168.1.8:9095
[Sarama] 2015/07/13 02:12:13 client/metadata fetching metadata for all topics from broker 192.168.1.8:9095
[Sarama] 2015/07/13 02:22:13 client/metadata fetching metadata for all topics from broker 192.168.1.8:9095
[Sarama] 2015/07/13 04:05:10 [backend/30dccfbe8daf] Triggering rebalance due to consumer list change
[Sarama] 2015/07/13 04:05:10 [backend/30dccfbe8daf] VeraTopic/3 :: Stopping partition consumer at offset -1
[Sarama] 2015/07/13 04:05:10 [backend/30dccfbe8daf] VeraTopic/0 :: Stopping partition consumer at offset -1
[Sarama] 2015/07/13 04:05:10 [backend/30dccfbe8daf] VeraTopic/1 :: Stopping partition consumer at offset -1
[Sarama] 2015/07/13 04:05:10 [backend/30dccfbe8daf] VeraTopic/2 :: Stopping partition consumer at offset 25199
[Sarama] 2015/07/13 04:05:10 client/metadata fetching metadata for all topics from broker 192.168.1.8:9095
[Sarama] 2015/07/13 04:05:10 consumer/broker/0 closed dead subscription to VeraTopic/3
[Sarama] 2015/07/13 04:05:10 consumer/broker/2 closed dead subscription to VeraTopic/1
[Sarama] 2015/07/13 04:05:10 consumer/broker/1 closed dead subscription to VeraTopic/0
[Sarama] 2015/07/13 04:05:10 consumer/broker/3 closed dead subscription to VeraTopic/2
[Sarama] 2015/07/13 04:05:11 [backend/30dccfbe8daf] VeraTopic :: Stopped topic consumer
[Sarama] 2015/07/13 04:05:12 [backend/30dccfbe8daf] Currently registered consumers: 0
[Sarama] 2015/07/13 04:05:12 [backend/30dccfbe8daf] VeraTopic :: Started topic consumer
panic: runtime error: integer divide by zero
[signal 0x8 code=0x7 addr=0xfa3af pc=0xfa3af]

goroutine 120 [running]:
runtime.panic(0x407d80, 0x6a8df9)
	/usr/local/go/src/pkg/runtime/panic.c:279 +0xf5
github.com/wvanbergen/kafka/consumergroup.dividePartitionsBetweenConsumers(0x6ad198, 0x0, 0x0, 0xc20865a100, 0x4, 0x4, 0x0)
	/Users/jness/v/gocode/src/github.com/wvanbergen/kafka/consumergroup/utils.go:39 +0x15f
github.com/wvanbergen/kafka/consumergroup.(*ConsumerGroup).topicConsumer(0xc2080523c0, 0x475c10, 0x9, 0xc208011b00, 0xc208012400, 0xc208406420)
	/Users/jness/v/gocode/src/github.com/wvanbergen/kafka/consumergroup/consumer_group.go:310 +0x5e1
created by github.com/wvanbergen/kafka/consumergroup.(*ConsumerGroup).topicListConsumer
	/Users/jness/v/gocode/src/github.com/wvanbergen/kafka/consumergroup/consumer_group.go:260 +0x340</pre>

--- a/consumergroup/consumer_group.go
+++ b/consumergroup/consumer_group.go
@@ -333,10 +333,15 @@ func (cg *ConsumerGroup) partitionConsumer(topic string, partition int32, messag
 	default:
 	}
 
-	err := cg.instance.ClaimPartition(topic, partition)
-	if err != nil {
-		cg.Logf("%s/%d :: FAILED to claim the partition: %s\n", topic, partition, err)
-		return
+	for maxRetries, tries := 3, 0; tries < maxRetries; tries++ {
+		if err := cg.instance.ClaimPartition(topic, partition); err == nil {
+			break
+		} else if err == kazoo.ErrPartitionClaimedByOther && tries+1 < maxRetries {
+			time.Sleep(1 * time.Second)
+		} else {
+			cg.Logf("%s/%d :: FAILED to claim the partition: %s\n", topic, partition, err)
+			return
+		}
 	}
 	defer cg.instance.ReleasePartition(topic, partition)
 
@@ -358,12 +363,12 @@ func (cg *ConsumerGroup) partitionConsumer(topic string, partition int32, messag
 	}
 
 	consumer, err := cg.consumer.ConsumePartition(topic, partition, nextOffset)
-	if (err == sarama.ErrOffsetOutOfRange) {
+	if err == sarama.ErrOffsetOutOfRange {
 		cg.Logf("%s/%d :: Partition consumer offset out of Range.\n", topic, partition)
 		// if the offset is out of range, simplistically decide whether to use OffsetNewest or OffsetOldest
 		// if the configuration specified offsetOldest, then switch to the oldest available offset, else
-		// switch to the newest available offset. 
-		if (cg.config.Offsets.Initial == sarama.OffsetOldest) {
+		// switch to the newest available offset.
+		if cg.config.Offsets.Initial == sarama.OffsetOldest {
 			nextOffset = sarama.OffsetOldest
 			cg.Logf("%s/%d :: Partition consumer offset reset to oldest available offset.\n", topic, partition)
 		} else {

